src/arrow_reader.rs

```diff
@@ -634,6 +677,9 @@ impl Decoder {
                     append_struct_null(idx, filed, builder, &iter.decoders[idx])?;
                 }
             }
+            &Decoder::List(_) => {
+                unreachable!("unreachable append null list")
```

Comment:
Why is this unreachable?

Reply:
I'm not sure here. I will add some code here and more tests in the following PRs.